src/session_concurrency/payload_dataset_manager.cc

```diff
@@ -86,9 +87,7 @@ PayloadDatasetManager::GetDelayForPayload(size_t dataset_index) const
   }
 
   if (!delay_ms_tensor_data.data_ptr) {
-    throw std::runtime_error(
-        "Missing 'delay' input for payload with index " +
-        std::to_string(dataset_index));
+    return {};
```

Comment:
As mentioned, GetDelay() now returns an optional in the case that the delay is missing, rather than throwing an exception.

As mentioned, GetDelay is now called before the request is sent in order to record request inputs for the profile export JSON. I.e. it's called _before_ the code knows if there's a delay for a particular request.

It's easier to return an optional for both cases.

The new place where the code expects a delay and errors if it's missing is here:

https://github.com/triton-inference-server/perf_analyzer/pull/302/files#diff-364ae5c05536cf1350b7890278a35b571113c32da7ca5f465d503a78ac996031R168-R172

The new place where